partner_multi_relation/models/res_partner.py

```diff
@@ -184,11 +184,10 @@ def action_view_relations(self):
                 'partner_multi_relation.action_res_partner_relation_all'
             ).read()[0]
             action['domain'] = [('id', 'in', relation_ids.ids)]
-            action['context'].\
-                update({'search_default_this_partner_id': contact.id,
-                        'default_this_partner_id': contact.id,
-                        'active_model': 'res.partner',
-                        'active_id': contact.id,
-                        'active_ids': [contact.id],
-                        'active_test': False})
+            action['context'] = {'search_default_this_partner_id': contact.id,
```

Comment:
This will override anything in the context, if there is a context. (The fields context is required in ir_act_window, so we can assume that there is at least an empty context: '{}'.

So safe_eval the existing context, and then do the update. The existing error is caused by the fact that context is not a dictionary, but a string containing a dictionary.

Reply:
You can't `safe_eval`, as it can contain JS variable names, and thus it will give error. You have to use a trick with `replace`:
```python
action['context'] = action['context'].replace('}', '<injected_vals>}')
```

Reply:
Thank you both for your suggestions. I'll work this and get a commit soon. I'm not quite sure how the python formatting will work with the '<injected_vals>' portion (it's not liking the quotations) but i'll continue to mess with it.

Reply:
Here you can see a more sophisticated example of some dictionary entries injection:

https://github.com/OCA/commission/blob/05a05f3060c2085f014ec231296e51659e675b59/sale_commission/models/sale_order.py#L43-L46

Reply:
I'm struggling to get this working, I understand the concept (I think) but do you have any other examples you can share?

Reply:
@pedrobaeza My latest commit (dd72e05) is my attempt at using your sample to update the context. It seems like it is updating it but something isn't right cause it gives the following error: 

Error: NameError: name 'contact' is not defined

I feel like it's close, feels like i'm missing something obvious.